File: 0520-Detect Capital/0520-Detect Capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        bool allUpper = false;
        bool allLower = false;
        if (islower(word[0]))
        {
            allLower = true;
        }
        
        for (int i = 1; i < word.size(); ++i)
        {
            if (islower(word[i]))
            {
                if (allUpper)
                {
                    return false;
                }
                    
                allLower = true;
            }
            else
            {
                if (allLower)
                {
                    return false;
                }
                    
                allUpper = true;
            }
        }
        
        return true;
    }
};

File: 0501-0600/0520-Detect Capital/0520-Detect Capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        return word.isupper() or word.islower() or word.istitle()

class Solution2:
    def detectCapitalUse(self, word: str) -> bool:
        allUpper = False
        allLower = word[0].islower()
        for c in word[1:]:
            if c.islower():
                if allUpper:
                    return False
                allLower = True
            else:
                if allLower:
                    return False
                allUpper = True
        return True
